serverless/serverless
# Store .gitignore as gitignore in template dir. Rename on template creation

## What did you implement:

Closes #3051

## How did you implement it:

Looks like this is a very common misconception people have about `npm publish` - https://github.com/npm/npm/issues/3763 . If there's no .npmignore file, but there is a .gitignore file, npm will rename it to .npmignore. 

So I renamed all `.gitignore`s in templates to just `gitignore`, so they will be left untouched by `npm pack`. And then I rename them back after copying template source directory.

> The common solution seems to be naming the .gitignore to some other name in your template files (I used gitignore) and having your scaffold cli tool rename it to .gitignore

## How can we verify it:

Before: 
```bash
$ npm pack
$ tar --list -f serverless-1.5.0.tgz | grep ignore
package/lib/plugins/create/templates/aws-csharp/.npmignore
package/lib/plugins/create/templates/aws-java-gradle/.npmignore
....
```

After:
```bash
$ npm pack
$ tar --list -f serverless-1.5.0.tgz | grep ignore
package/lib/plugins/create/templates/aws-csharp/gitignore
package/lib/plugins/create/templates/aws-java-gradle/gitignore
...
```

And more real-life check:

```bash
$ serverless create --template aws-nodejs
$ ls -a
.  ..  .gitignore  handler.js  serverless.yml
```

So there must be `.gitignore` instead of `.npmignore`

## Todos:



***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
